rest_framework/static/rest_framework/js/ajax-form.js

```diff
@@ -37,6 +37,19 @@ function doAjaxSubmit(e) {
 
   if (contentType) {
     data = form.find('[data-override="content"]').val() || ''
+
+    if (contentType === 'multipart/form-data') {
+      // We need to add a boundary parameter to the header
+      // We assume the first valid-looking boundary line in the body is correct
+      // regex is from RFC 2046 appendix A
+      var re = /^--([0-9A-Z'()+_,-./:=?]{1,70})[ \f\t\v\u00a0\u1680\u180e\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]*$/im;
```

Comment:
- Now handles space in the boundary delimiter correctly (with a more obvious translation from the spec)
- Simplified whitespace to `\s*?` (the non-greedy qualifier because we want the `$` to match the current line only and `\s` includes `\n`)

Re-tested locally with and without whitespace in the boundary; django handles this correctly too.